docs/controllers/cluster.md

```diff
@@ -0,0 +1,27 @@
+# Cluster controller
+
+## Overview
+
+Cluster controller is responsible managing Cluster and InfrastructureCluster(AWSCluster, etc.) CRs. Both object will
+represent the current cluster where operator is running because we treat this cluster as both management and workload.
+
+## Behavior
+
+```mermaid
+stateDiagram-v2
+    [*] --> IsCurrentPlatformSupported
+    state IsCurrentPlatformSupported <<choice>>
+    IsCurrentPlatformSupported --> NoOp: False
+    IsCurrentPlatformSupported --> CreateOrUpdateInfrastructureCluster: True
+    CreateOrUpdateInfrastructureCluster --> SetInfrastructureClusterSpec
+    SetInfrastructureClusterSpec --> SetInfrastructureClusterStatusReady
+    SetInfrastructureClusterStatusReady --> CreateOrUpdateCoreCluster
+    CreateOrUpdateCoreCluster --> SetCoreClusterInfrastructureRef
+    SetCoreClusterInfrastructureRef --> [*]
+    NoOp --> [*]
+```
+
+If the current platform is not supported, the controller will not create any cluster objects and allow BYO scenarios.
```

Comment:
i think we should just say "bring your own" instead of BYO. you could even add quotes around it to denote its meaning. eg
```suggestion
If the current platform is not supported, the controller will not create any cluster objects and allow "bring your own" scenarios.
```